### PR TITLE
HUD: InspectMenu rework

### PR DESCRIPTION
**Pull request by Alf21**
Fixed continous netstream (replaced by RPC triggers) and improved design.

image

After pressing `E`:
image

### CODE REVIEW

File: code/player/Player.cs [hunk 1]
```diff
@@ -242,14 +255,17 @@ public override void TakeDamage(DamageInfo info)
                 info.Damage *= 2.0f;
             }
 
-            if (info.Attacker is TTTPlayer attacker && attacker != this)
+            using (Prediction.Off())
```
Comment (mzegar) [inline code/player/Player.cs:258]: We do not want to turn off prediction here I'm pretty sure. If the FP guys didn't, we shouldn't.

https://wiki.facepunch.com/sbox/Prediction

File: code/player/Player.cs [hunk 2]
```diff
@@ -189,30 +193,39 @@ private void TickPlayerDropWeapon()
...
                     }
 
+                    return;
+                }
+
+                if (inspectingPlayerCorpse != null)
+                {
                     ClientCloseInspectMenu(client);
+
+                    inspectingPlayerCorpse = null;
```
Comment (mzegar) [inline code/player/Player.cs:228]: Is there any cases where the inspect menu fails to close?

File: code/ui/InspectMenu.cs [hunk 3]
```diff
@@ -21,33 +22,161 @@ public bool IsShowing
...
-                    + $"{deadPlayer.Role.Name}";
+                confirmationPanel.SetClass("hide", true);
+
+                header.SetPlayer(deadPlayer);
+                header.SetClass("hide", false);
+
+                content.SetPlayer(deadPlayer);
+                content.SetClass("hide", false);
+
+                footer.SetClass("hide", false);
```
Comment (mzegar) [inline code/ui/InspectMenu.cs:66]: Is there a way we can setup this UI to be a single element that gets "hidden"/"unhidden"?

**Review by mzegar — COMMENTED** [review]
Comments

File: code/player/Player.cs (hunk 2 above)
Comment (Alf21) [reply code/player/Player.cs:228]: not in my tests, but feel free trying some edge-cases :D

File: code/ui/InspectMenu.cs (hunk 3 above)
Comment (Alf21) [reply code/ui/InspectMenu.cs:66]: What do you mean exactly? Currently, it's one UI element, that includes more or less 4 panels, 3 of them should just be shown if the corpse was confirmed. The other, if not confirmed. I think that can be handled better as well (just 2 panels), if that's what you meant?

Comment (mzegar) [reply code/ui/InspectMenu.cs:66]: If the 3 panels could have one single "hide" call made on them. As in compile the 3 panels into a single "component", then just "hide' the single compontent.

Comment (Alf21) [reply code/ui/InspectMenu.cs]: Done

**Review by mzegar — APPROVED** [review]
LGTM
